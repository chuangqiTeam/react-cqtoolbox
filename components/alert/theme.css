@import "../colors.css";
@import "../variables.css";

:root {
  --alert-hasDscription-icon-font-size: calc(2.4 * var(--unit));
  --alert-hasDscription-padding: var(--pd) calc(4.8 * var(--unit)) var(--pd) calc(6 * var(--unit));
  --alert-noIocn-padding: var(--pd) calc(4.8 * var(--unit)) var(--pd) calc(1.6 * var(--unit));
  --alert-padding: var(--pd) calc(4.8 * var(--unit)) var(--pd) calc(3.8 * var(--unit));
  --lpd: calc(1.6 * var(--unit));
  --pd: calc(.8 * var(--unit));
}

.alert {
  font-size: var(--font-size);

  position: relative;

  overflow: hidden;

  opacity: 1;
  border-radius: 4px;
  @apply --reset;
}

.hasDscription {
  .alertInner {
    padding: var(--alert-hasDscription-padding);
  }
  .icon {
    font-size: var(--alert-hasDscription-icon-font-size);
    left: var(--lpd);
  }

  .message {
    margin-bottom: var(--pd);
  }
}

.noIcon {
  .alertInner {
    padding: var(--alert-noIocn-padding);
  }
}

.success {
  border: 1px solid var(--palette-green-100);
  background-color: var(--palette-green-50);

  .message {
    color: var(--palette-green-800);
  }

  .icon {
    color: var(--palette-green-600);
  }
}

.info {
  border: 1px solid var(--palette-cyan-100);
  background-color: var(--palette-cyan-50);

  .message {
    color: var(--palette-cyan-800);
  }

  .icon {
    color: var(--palette-cyan-600);
  }
}

.warning {
  border: 1px solid var(--palette-amber-100);
  background-color: var(--palette-amber-50);

  .message {
    color: var(--palette-amber-800);
  }

  .icon {
    color: var(--palette-amber-600);
  }
}

.error {
  border: 1px solid var(--palette-red-100);
  background-color: var(--palette-red-50);

  .message {
    color: var(--palette-red-800);
  }

  .icon {
    color: var(--palette-red-600);
  }
}

.alertInner {
  padding: var(--alert-padding);
}

.message {
  display: block;
}

.description {
  font-size: var(--font-size-small);
  line-height: 1.75;

  color: var(--palette-grey-600);
}

.icon {
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  left: var(--lpd);
}

.closeIcon {
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: var(--lpd);
  cursor: pointer;
}

.anim {
  transition: all var(--animation-duration) var(--animation-curve-linear-out-slow-in);

  opacity: 0;
}
