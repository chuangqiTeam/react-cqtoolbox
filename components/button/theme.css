@import "../colors.css";
@import "../variables.css";
@import "./config.css";

.buttonGroup {
  & .squared {
    border-radius: 0;
    &:first-child {
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
    }

    &:last-child {
      border-top-right-radius: 4px;
      border-bottom-right-radius: 4px;
    }
  }

  & .neutral:not([disabled]).flat,
  & .primary:not([disabled]).flat,
  & .accent:not([disabled]).flat {
    border-width: 1px 0 1px 1px;
    border-style: solid;

    &:last-child {
        border-right-width: 1px;
    }
  }
}

.button {
  font-size: var(--font-size);
  font-weight: normal;

  position: relative;

  display: inline-block;

  padding: 0;

  cursor: pointer;
  transition: box-shadow .2s var(--animation-curve-fast-out-linear-in),
  background-color .2s var(--animation-curve-default),
  color .2s var(--animation-curve-default);
  text-align: center;
  white-space: nowrap;
  text-decoration: none;
  letter-spacing: 0;
  text-transform: uppercase;

  border: 0;
  outline: none;

  @apply --reset;

  &::-moz-focus-inner {
    border: 0;
  }


  & > * {
    pointer-events: none;
  }

  & > .rippleWrapper {
    overflow: hidden;
  }

  &[disabled] {
    cursor: auto;
    pointer-events: none;

    color: var(--button-disabled-text-color);
  }
}

.normal {
  line-height: var(--button-normal-line-height);
}

.small {
  line-height: var(--button-small-line-height);
}

.large {
  line-height: var(--button-large-line-height);
}

.squared {
  min-width: var(--button-squared-min-width);
  padding: var(--button-squared-padding);

  border-radius: var(--button-border-radius);

  & .icon {
    margin-right: var(--button-squared-icon-margin);

    vertical-align: baseline;
  }
}


.raised {
  composes: button;
  composes: squared;
  &[disabled] {
    border: 1px solid var(--button-disabled-background-color);
    background-color: var(--button-disabled-background-color);
  }
}

.flat {
  background: transparent;

  composes: button;
  composes: squared;
  &[disabled] {
    border: 1px solid transparent;
  }
}

.floating {
  font-size: var(--button-floating-font-size);

  width: var(--button-floating-height);
  height: var(--button-floating-height);

  border-radius: 50%;
  box-shadow: 0 1px 1.5px 0 rgba(0, 0, 0, .12),
  0 1px 1px 0 rgba(0, 0, 0, .24);

  composes: button;

  & .icon:not([data-react-toolbox="tooltip"]) {
    line-height: var(--button-floating-height);
  }

  & > .rippleWrapper {
    border-radius: 50%;
  }

  &.mini {
    font-size: var(--button-floating-mini-font-size);

    width: var(--button-floating-mini-height);
    height: var(--button-floating-mini-height);

    & .icon {
      line-height: var(--button-floating-mini-height);
    }
  }
}

.toggle {
  width: var(--button-height);

  vertical-align: middle;

  border-radius: 50%;
  background: transparent;

  composes: button;

  & > .icon,
  & > .rippleWrapper {
    border-radius: 50%;
  }
}

.primary:not([disabled]) {
  &.raised,
  &.floating {
    color: var(--button-primary-color-contrast);
    border: 1px solid var(--button-primary-color);
    background: var(--button-primary-color);
  }

  &.flat,
  &.toggle {
    color: var(--button-primary-color);
    border: 1px solid var(--button-primary-border-color);
    &:focus:not(:active) {
      background: var(--button-primary-color-hover);
    }
  }

  &.flat:hover {
    background: var(--button-primary-color-hover);
  }
}

.accent:not([disabled]) {
  &.raised,
  &.floating {
    color: var(--button-accent-color-contrast);
    border: 1px solid var(--button-accent-color);
    background: var(--button-accent-color);
  }

  &.flat,
  &.toggle {
    color: var(--button-accent-color);
    border: 1px solid var(--button-accent-border-color);

    &:focus:not(:active) {
      background: var(--button-accent-color-hover);
    }
  }

  &.flat:hover {
    background: var(--button-accent-color-hover);
  }
}

.neutral:not([disabled]) {
  &.raised,
  &.floating {
    color: var(--button-neutral-color-contrast);
    border: 1px solid var(--button-neutral-color);
    background-color: var(--button-neutral-color);
  }

  &.flat,
  &.toggle {
    color: var(--button-neutral-color-contrast);
    border: 1px solid var(--button-neutral-border-color);

    &:focus:not(:active) {
      background: var(--button-neutral-color-hover);
    }
  }

  &.flat:hover {
    background: var(--button-neutral-color-hover);
  }
}
