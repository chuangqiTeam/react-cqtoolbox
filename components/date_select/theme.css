.DangePicker {
    display: inline-flex;
    user-select: none;
}
