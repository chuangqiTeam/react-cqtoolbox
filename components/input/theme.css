@import '../colors.css';
@import '../variables.css';
@import './config.css';

.input {
    display: flex;
    align-items: center;
    position: relative;
    @apply --reset;
    font-size: var(--font-size);
    padding: var(--input-field-padding) 0;
    border-bottom: 1px solid var(--input-text-bottom-border-color);
}

.inputElement {
    appearance: none;
    background-image: none;
    background-color: var(--input-text-background-color);
    color: var(--input-color);
    box-sizing: border-box;
    font-size: inherit;
    padding: 0 var(--input-field-padding);
    line-height: inherit;
    border: 0;
    outline: none;
    width: 100%;
    flex: 1;

    &:focus:not([disabled]):not([readonly]) {
        & ~ .bar::before,
        & ~ .bar::after {
          width: 50%;
        }
    }
}

.icon {
    font-size: var(--font-size-large);
}

.bar {
  display: block;
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;

  &::before,
  &::after {
    background-color: var(--input-text-highlight-color);
    bottom: 0;
    content: '';
    height: 2px;
    position: absolute;
    transition-duration: 0.2s;
    transition-property: width, background-color;
    transition-timing-function: var(--animation-curve-default);
    width: 0;
  }

  &::before {
    left: 50%;
  }

  &::after {
    right: 50%;
  }
}

.normal {
    line-height: var(--input-line-height);

    .prefix, .suffix {
        width: var(--input-icon-width);
        height: var(--input-icon-width);
        line-height: var(--input-icon-width);
    }
}

.small {
    line-height: var(--input-small-line-height);

    .prefix, .suffix {
        width: var(--input-small-icon-width);
        height: var(--input-small-icon-width);
        line-height: var(--input-small-icon-width);
    }
}

.large {
    line-height: var(--input-large-line-height);

    .prefix, .suffix {
        width: var(--input-large-icon-width);
        height: var(--input-large-icon-width);
        line-height: var(--input-large-icon-width);
    }
}

.prefix, .suffix {
    color: var(--input-icon-color);
    transition: color 0.2s var(--animation-curve-default);
    cursor: pointer;

    &.search:hover {
        color: var(--input-border-color-focus);
    }

    &.close-circle {
        color: var(--input-text-bottom-border-color);
    }

    &.close-circle:hover {
        color: var(--input-icon-close-color-hover);
    }
}



.textarea {
    display: block;
    width: 100%;
    resize: vertical;
    padding: 5px 7px;
    line-height: var(--input-line-height);

    @apply --reset;

    font-size: var(--font-size);
    border: 1px solid var(--input-text-bottom-border-color);
    color: var(--input-color);
    background-color: #fff;
    background-image: none;
    border-radius: 4px;
    transition: border-color var(--animation-duration) var(--animation-curve-default);

    &:focus:not([disabled]):not([readonly]),
    &:hover:not([disabled]):not([readonly]) {
        outline: none;
        border-color: var(--input-border-color-focus);
    }
}
