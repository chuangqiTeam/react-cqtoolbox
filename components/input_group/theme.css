@import "../colors.css";
@import "../variables.css";

:root {
  --input-group-large-padding: calc(.8 * var(--unit));
  --input-group-normal-padding: calc(.4 * var(--unit));
  --input-text-border-color: color(var(--color-black) a(12%));
}

.inputGroup {
  display: flex;

  & > :first-child {
    border-top-left-radius: 4px;
    border-bottom-left-radius: 4px;
  }

  & > * {
    border-radius: 0;
  }

  & > :last-child {
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
  }

  & .select_input {
    min-width: auto;
  }

  & .input,
  & .select_input,
  & .neutral:not([disabled]).flat {
    border-width: 1px 0 1px 1px;
    border-style: solid;
    border-color: var(--input-text-border-color);

    &:last-child {
      border-right-width: 1px;
    }
  }

  & .input {
    flex: 1;
  }

  & .bar {
    display: none;
  }
}

.small {
  & .input {
    padding: 0;
  }
}

.normal {
  & .input {
    padding: var(--input-group-normal-padding);
  }
}

.large {
  & .input {
    padding: var(--input-group-large-padding);
  }
}
