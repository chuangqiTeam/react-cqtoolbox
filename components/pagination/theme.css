@import "../variables.css";

.pagination {
    margin: var(--unit) 0;
}

.space {
    margin-left: var(--unit);
}
