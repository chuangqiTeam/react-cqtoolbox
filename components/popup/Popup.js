//flow

import React from 'react';
import { findDOMNode } from 'react-dom';
import domAlign from 'dom-align';
import classnames from 'classnames';
import events from '../utils/events.js';
import Overlay from '../overlay';

type Theme = {
  overlay: string,
  popup: string,
  active: string,
}

type DefaultProps = {
  mask: boolean,
  active: boolean,
  matchTargetWidth: boolean,
}

type Props = {
  // 是否激活
  active: boolean,
  // 位置
  align: any,
  // 是否显示浮层
  mask: boolean,

  children: React.Element<*>,
  // 是否宽度等于目标元素
  matchTargetWidth: boolean,

  onMouseEnter: () => void,

  onMouseLeave: () => void,
  // 获得目标元素
  getRootDomNode: () => void,
  // 请求关闭popup
  onRequestClose: () => void,

  theme: Theme,
}

class Popup extends React.Component<DefaultProps, Props, any> {

  static defaultProps = {
    mask: false,
    active: false,
    matchTargetWidth: false,
  }

  componentDidMount() {
    const { active, align, matchTargetWidth, getRootDomNode } = this.props;
    const source = this.getPopupDomNode();
    const target = getRootDomNode && getRootDomNode();

    if (target) {
      // set popup width
      const widthProp = matchTargetWidth ? 'width' : 'minWidth';
      source.style[widthProp] = `${target.offsetWidth}px`;

      // set popup position
      this.setPopupAlign(source, target, align);
    }

    if (active) {
      this.clickOutsideHandler = true;
      events.addEventsToDocument([{
        click: this.onDocumentClick,
      }]);
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.active && !this.clickOutsideHandler) {
      this.clickOutsideHandler = true;
      events.addEventsToDocument({
        click: this.onDocumentClick,
      });
    }
  }

  componentWillUnmount() {
    this.removeEventsFromDocument();
  }

  removeEventsFromDocument = () => {
    this.clickOutsideHandler = false;
    events.removeEventsFromDocument({
      click: this.onDocumentClick,
    });
  }

  setPopupAlign = (sourceNode, targetNode, popupAlign) => {
    domAlign(sourceNode, targetNode, popupAlign);
  }

  onDocumentClick = (event) => {
    if (this.props.mask && !this.props.maskClosable) {
      return;
    }

    const rootNode = this.props.getRootDomNode();
    const popupNode = this.getPopupDomNode();

    if (!events.targetIsDescendant(event, rootNode) &&
    !events.targetIsDescendant(event, popupNode)) {
      this.props.onRequestClose(false);
    }
  }

  getMaskElement = () => {
    const {theme, mask, active, onOverlayClick} = this.props;

    return mask ?
    <Overlay
      onClick={onOverlayClick}
      active={active}
      className={theme.overlay} /> : null;
  }

  getPopupElement = () => {
    const {children, theme, active, onMouseEnter, onMouseLeave} = this.props;

    const newProps = {
      onMouseEnter,
      onMouseLeave,
    };

    const classes = classnames(theme.popup, {
      [theme.active]: active});

    return <div ref="popup" className={classes}>{React.cloneElement(children, newProps)}</div>;
  }

  getPopupDomNode = () => {
    return findDOMNode(this.refs.popup);
  }

  render () {
    return (
      <div>
        {this.getMaskElement()}
        {this.getPopupElement()}
      </div>
    );
  }
}

export default Popup;
