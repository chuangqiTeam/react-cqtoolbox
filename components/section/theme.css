@import "../colors.css";
@import "../variables.css";
@import "./config.css";


.section {
  padding: var(--section-padding);

  border-radius: 2px;
  background-color: var(--section-background-color);
  box-shadow: 0 1px 2px rgba(0, 0, 0, .2);

  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.content {
    &.center {
        display: flex;
        align-items: center;
        justify-content: center;
        margin: calc(3 * var(--unit)) 0;
    }
}


.title {
  position: relative;

  display: flex;

  margin-bottom: var(--section-title-margin-bottom);
  padding-bottom: 10px;

  border-bottom: 1px solid #f1f1f1;

  align-items: center;
  justify-content: space-between;
}

.titleInner {
  display: inline-flex;

  align-items: center;
  & > .tip {
    width: 4px;
    margin-right: 6px;

    background-color: var(--color-primary);
  }

  & > h2 {
    margin: 0;

    color: var(--section-title-color);
  }

  &.normal {
    & > h2 {
      font-size: var(--section-title-font-size);
    }

    & > .tip {
      height: var(--section-title-font-size);
    }
  }

  &.small {
    & > h2 {
      font-size: var(--section-subtitle-font-size);
    }

    & > .tip {
      height: var(--section-subtitle-font-size);
    }
  }
}

.nullText {
    font-size: var(--font-size-large);
    color: var(--color-text-secondary);
    text-align: center;
}
