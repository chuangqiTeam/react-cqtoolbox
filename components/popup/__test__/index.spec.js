import React from 'react';
import { mount } from 'enzyme';
import Popup from '../popup';
import theme from '../theme.css';



describe('Popup', () => {

  describe('#render', () => {

    it('渲染正确', () => {
      const wrapper = mount(
        <Popup theme={theme}>
          <span>浮层</span>
      </Popup>);
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('渲染出mask浮层', () => {
      const wrapper = mount(
        <Popup mask theme={theme}>
          <span>浮层</span>
      </Popup>);
      expect(wrapper.render()).toMatchSnapshot();
    });

  });
})
