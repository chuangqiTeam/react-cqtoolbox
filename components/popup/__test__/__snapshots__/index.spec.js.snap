// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup #render 渲染出mask浮层 1`] = `
<div>
  <div
    class="overlay"
    data-react-toolbox="overlay"
  />
  <div
    class="popup"
  >
    <span>
      浮层
    </span>
  </div>
</div>
`;

exports[`Popup #render 渲染正确 1`] = `
<div>
  <div
    class="popup"
  >
    <span>
      浮层
    </span>
  </div>
</div>
`;
